tds.go

```diff
@@ -1250,7 +1250,12 @@ initiate_connection:
 
 	if sess.routedServer != "" {
 		toconn.Close()
-		p.Host = sess.routedServer
+		var routedServer []string
+		routedServer = strings.Split(sess.routedServer, "\\")
```

Comment:
1. Add a comment to address why/when this may happen.
2. remove: `var routedServer []string`
3. use strings.SplitN for consistency with other code:
`routedParts := strings.SplitN(sess.routedServer, "\\", 2)`